java-frontend/src/main/java/org/sonar/java/se/xproc/MethodBehavior.java

```diff
@@ -156,11 +188,83 @@ public boolean isComplete() {
   public void completed() {
     this.complete = true;
     this.visited = true;
+    reduceYields();
+    System.out.println(signature);
+    yields.forEach(System.out::println);
+  }
+
+  private void reduceYields() {
+    Set<HappyPathYield> happyPathYields = happyPathYields().filter(y -> y.resultIndex() == -1).collect(Collectors.toCollection(LinkedHashSet::new));
+    yields.removeAll(happyPathYields);
+    int count = happyPathYields.size();
+    Set<HappyPathYield> newYields = reduce(happyPathYields);
+    while (newYields.size() < count) {
+      count = newYields.size();
+      newYields = reduce(newYields);
+    }
+    yields.addAll(newYields);
+  }
+
+  private Set<HappyPathYield> reduce(Set<HappyPathYield> yieldsToReduce) {
+    LinkedList<HappyPathYield> yields = new LinkedList<>(yieldsToReduce);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Rename "yields" which hides the field declared at line 48. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AHiddenFieldCheck)